src/app/shared/subscriptions/subscription-modal/subscription-modal.component.ts

```diff
@@ -0,0 +1,260 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+import { BehaviorSubject, combineLatest, from, shareReplay } from 'rxjs';
+import { map, mergeMap, take, tap } from 'rxjs/operators';
+import { TranslateService } from '@ngx-translate/core';
+import { NgbActiveModal, NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import findIndex from 'lodash/findIndex';
+
+import { Subscription } from '../models/subscription.model';
+import { DSpaceObject } from '../../../core/shared/dspace-object.model';
+import { SubscriptionsDataService } from '../subscriptions-data.service';
+import { NotificationsService } from '../../notifications/notifications.service';
+import { PaginatedList } from '../../../core/data/paginated-list.model';
+import { RemoteData } from '../../../core/data/remote-data';
+import { getFirstCompletedRemoteData, getFirstSucceededRemoteDataPayload } from '../../../core/shared/operators';
+import { AuthService } from '../../../core/auth/auth.service';
+import { isNotEmpty } from '../../empty.util';
+
+@Component({
+  selector: 'ds-subscription-modal',
+  templateUrl: './subscription-modal.component.html',
+  styleUrls: ['./subscription-modal.component.scss']
+})
+export class SubscriptionModalComponent implements OnInit {
```

Comment:
Please add comments or TypeDocs to describe the overall purpose of this component